### PR TITLE
feat: generate secret function and replaced few instances

### LINKED ISSUE #4588: Introduce APP_SECRET to replace SECRETS
## Context

We currently have 4 environment variables to specify our SECRETS in `packages/twenty-server/.env`:
- ACCESS_TOKEN_SECRET
- LOGIN_TOKEN_SECRET
- REFRESH_TOKEN_SECRET
- FILE_TOKEN_SECRET

We would like to replace those by a unique APP_SECRET env variable and to generate those 4 when needed as:
- ACCESS_TOKEN_SECRET=SHA256({APP_SECRET}+{workspaceId}+"ACCESS")
- LOGIN_TOKEN_SECRET=SHA256({APP_SECRET}+{workspaceId}+"FILE")
- ...

# What & How

- Remove those 4 secrets from the code base
- introduce APP_SECRET instead (this needs to be introduced in several places, I would recommend look for ACCESS_TOKEN_SECRET and to mimick it)
- compute accessTokenSecret based on the formula above whenever it's needed

### PR DESCRIPTION
**Pull request by Khaan25**
This PR fixes #4588

### CODE REVIEW

File: packages/twenty-server/src/utils/generate-secret.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { createHash } from 'crypto';
+
+if (!process.env.APP_SECRET) {
```
Comment (FelixMalfait) [inline packages/twenty-server/src/utils/generate-secret.ts:3]: We should access env var through the dedicate service not directly through process.env ; you should probably do this in AuthService

**Review by FelixMalfait — CHANGES_REQUESTED** [review]
Thank you! Please replace every instance in the codebase :)

File: packages/twenty-server/src/utils/generate-secret.ts (hunk 1 above)
Comment (Khaan25) [reply packages/twenty-server/src/utils/generate-secret.ts:3]: Got it. I'll see that.

**Comment by Khaan25** [comment]
Haha, no worries. I'm also learning about the new techniques and codebase 🙌 Amazing Job

**Comment by Khaan25** [comment]
I guess the PR is complete then? @FelixMalfait Thanks for the help though. I got the understanding of the codebase as well how things are linkedin togther, do you think we can close the PR now?

**Comment by FelixMalfait** [comment]
/award 300
Yes PR is complete, @charlesBochet  will review it again when he has time!

**Comment by Khaan25** [comment]
@FelixMalfait, hope I get points for the change as it's for developer docs as per side quest :)

Let me knowww

File: packages/twenty-website/src/content/developers/self-hosting/self-hosting-var.mdx [hunk 2]
```diff
@@ -51,7 +51,7 @@ yarn command:prod cron:calendar:calendar-event-list-fetch
 ### Tokens
 
 <ArticleTable options={[
-    ['APP_SECRET', '<random>', 'Secret used for encryption across the app'],
+    ['APP_SECRET', '<random>', 'Secret used for the app token'],
```
Comment (FelixMalfait) [inline packages/twenty-website/src/content/developers/self-hosting/self-hosting-var.mdx:54]: I think the previous message was more clear :)

File: packages/twenty-website/src/content/developers/self-hosting/cloud-providers.mdx [hunk 3]
```diff
@@ -271,7 +268,7 @@ resource "azapi_update_resource" "cors" {
 ```hcl
 # backend.tf
 
-# Create a random UUIDs
+# Create app random UUID
```
Comment (FelixMalfait) [inline packages/twenty-website/src/content/developers/self-hosting/cloud-providers.mdx:271]: Create a random UUID

File: packages/twenty-website/src/content/developers/self-hosting/self-hosting-var.mdx (hunk 2 above)
Comment (Khaan25) [reply packages/twenty-website/src/content/developers/self-hosting/self-hosting-var.mdx:54]: sure :)

File: packages/twenty-website/src/content/developers/self-hosting/cloud-providers.mdx [hunk 4]
```diff
@@ -335,10 +332,6 @@ resource "azurerm_container_app" "twenty_server" {
         value = "postgres://${local.db_user}:
         ${local.db_password}@${local.db_app_name}:5432/default"
       }
-      env {
```
Comment (FelixMalfait) [inline packages/twenty-website/src/content/developers/self-hosting/cloud-providers.mdx:341]: Why remove this?

Comment (Khaan25) [reply packages/twenty-website/src/content/developers/self-hosting/cloud-providers.mdx:341]: mistakenly removed :)

**Pull request merged**